selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/factories/CreateFactoryPage.java

```diff
@@ -167,10 +167,12 @@ public void waitErrorMessageNotVisible() {
     seleniumWebDriverHelper.waitInvisibility(By.xpath(Locators.NEW_FACTORY_ERROR_MESSAGE));
   }
 
-  public void waitUploadFileButton() {
-    seleniumWebDriverHelper.waitVisibilityAndGetEnableState(uploadFileButton);
+  public boolean isUploadFileButtonEnabled() {
+    return seleniumWebDriverHelper.waitVisibilityAndGetEnableState(uploadFileButton);
   }
 
+  // we have to use the hidden web-element, because the file is uploaded when we send the command to
```

Comment:
IMHO it's important to explain why we can't use `uploadFileButtn` directly, like the follow:
`// we can't use 'uploadFileButton' element directly here because it is a DIV which just calls hidden input of file type 'hiddenUploadInputButton'`